src/services/comment.service.ts

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { UseFormResetField } from 'react-hook-form';
+import { Action, ActionTypes } from '../reducer/store';
+import { addComment, deleteComment } from '../api/users.api';
+import { Comment, CommentData } from '../types/Comment';
+
+export function commentService(
+  dispatch: (action: Action) => void,
+  commentsList: Comment[] | null,
+) {
+  function addNewComment(
+    dataToSend: CommentData & { postId: number },
+    setShowSpinner: React.Dispatch<React.SetStateAction<boolean>>,
+    resetField: UseFormResetField<CommentData>,
+  ) {
+    addComment(dataToSend)
+      .then((res) => {
+        dispatch({
+          type: ActionTypes.getComments,
+          comments: commentsList ? [...commentsList, res] : [res],
+        });
+
+        resetField('body');
+      })
+      .finally(() => {
+        setShowSpinner(false);
+      });
+  }
+
+  function removeComment(id: number) {
+    const setComments = commentsList
```

Comment:
The delete operation should ideally be performed first and then the state should be updated based on the result of the delete operation. Currently, the comment is being removed from the state before the delete operation is performed, which could lead to inconsistencies if the delete operation fails.